azure-iot-device/azure/iot/device/common/transport/mqtt/mqtt_transport.py

```diff
@@ -488,6 +508,20 @@ def _get_input_topic_for_subscribe(self):
         """
         return self._get_topic_base() + "/inputs/#"
 
+    def _get_method_topic_for_subscribe(self):
+        """
+        :return: The topic for ALL incoming methods. It is of the format
+        "$iothub/methods/POST/#"
+        """
+        return "$iothub/methods/POST/#"
```

Comment:
>$iothub/methods/POST/# [](start = 16, length = 22)

there's little repetition so i'm not too worried about this but i think it'd be nice to have all topics grouped in a set of constants co-located together. for when we need to add/change/check a topic it's easier to know where it's at and how it compares to other topics. also a nice self-documenting thing.

Reply:
The issue is not all are constants. Most are dynamically created. That said, I think there's an argument for defining constant format strings with the dynamic keywords, as in the function below.

---
In reply to: [280650160](https://github.com/Azure/azure-iot-sdk-python-preview/pull/86#discussion_r280650160) [](ancestors = 280650160)